firmware-management/mbl-watchdog-init/mbl-watchdog-init.c

```diff
@@ -120,6 +120,12 @@ int numeric_string_to_positive_int(const char* const str)
     char *endptr;
     const long int res = strtol(str, &endptr, 0);
 
+    if (*endptr != '\0' || endptr == str)
```

Comment:
If `strtol` fails I'm not sure that we can assume that `endptr` is valid, so I'd put this check after the `errno != 0 && res == 0` check below.